android/src/play/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -676,4 +675,77 @@ private void sendEvent(
         .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
         .emit(eventName, params);
   }
+
+  @ReactMethod
+  public void isSubscribed(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchaseHistoryAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        for (int i = 0; i < list.size(); i++) {
+          PurchaseHistoryRecord purchase = list.get(i);
+          if (purchase.getSkus().get(0).equalsIgnoreCase(sku)) {
+            promise.resolve(true);
+            return;
+          }
+        }
+        promise.resolve(false);
+      });
+    });
+  }
+
+
+  @ReactMethod
+  public void getSubscriptionTransactionDetails(final String sku, final Promise promise) {
+    ensureConnection(promise, billingClient -> {
+      billingClient.queryPurchasesAsync( BillingClient.SkuType.SUBS, (billingResult, list) -> {
+        WritableMap item = Arguments.createMap();
+        Purchase purchase = list.get(0);
```

Comment:
just to be safe you could surround this with a try catch, if list != null but doesn't have any elements you can get a `IndexOutOfBoundsException`